cluster/swarm/utils.go

```diff
@@ -33,3 +34,13 @@ func convertMapToKVStrings(values map[string]*string) []string {
 	}
 	return result
 }
+
+var imageEngineOSErrorPattern = regexp.MustCompile(`cannot load (.+) image on (.+)`)
```

Comment:
Seems like this specific text comes only from OS mismatch errors: https://github.com/moby/moby/blob/master/image/tarexport/load.go#L428

This error could change and break this issue, but the engine error text is also somewhat ad-hoc, so this is probably the best we can do.